ChatRPG/ChatRPG/Data/ApplicationDbContext.cs

```diff
@@ -28,11 +28,16 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         base.OnModelCreating(modelBuilder);
 
         modelBuilder
-            .Entity<CharacterAbility>(builder => builder.HasKey(c => new { c.CharacterId, c.AbilityId }))
-            .Entity<CharacterLocation>(builder =>
+            .Entity<CharacterAbility>(builder =>
+            {
+                builder.HasKey(c => new { c.CharacterId, c.AbilityId });
+                builder.HasOne(c => c.Character).WithMany(c => c.CharacterAbilities);
+            })
+            .Entity<CharacterEnvironment>(builder =>
```

Comment:
Kan du ikke forklare mig hvad der foregår her i vc sometime?

Reply:
Don't even worry about it

Reply:
JK!
Entity Framework kan automatisk "infer" en hel masse om din datamodel, f.eks. dens primary keys, foreign keys, indices, datatyper, constraints, og så videre.
Men i nogle tilfælde kan den ikke automatisk vælge noget (kan være din datamodel er "tvetydig"), eller også vil du bare selv have kontrol over, hvad EF genererer ud fra din datamodel. 
I `ApplicationDbContext` kan du definere dine egne regler for genereringen af datamodellen. 
Helt konkret, så ved EF ikke hvilken primary key den skal bruge i `CharacterAbility`. Derfor definerer vi her dens PK som en composite key - syntaksen er bare sådan det skal se ud i EF, hvis du følger deres dokumentation.

Derudover hjælper vi også EF med relationerne her, ved at sige at en `CharacterAbility` "HasOne" `Character`, mens `Character` "HasMany" `CharacterAbility` -> med andre ord definerer vi et one-to-many forhold, og fortæller EF hvilke properties den skal bruge til at realisere forholdet.